source/extensions/filters/http/aws_lambda/sts_credentials_provider.cc

```diff
@@ -113,21 +114,34 @@ void StsCredentialsProviderImpl::onFailure(CredentialsFailureStatus status,
   }
 }
 
+// for a lambda call check for credentials on the given role_arn_arg,
+// if not the same role as the default one then chain assuming roles
+// unless explicitly told not to.
 StsConnectionPool::Context *StsCredentialsProviderImpl::find(
     const absl::optional<std::string> &role_arn_arg,
+    bool disable_role_chaining,
     StsConnectionPool::Context::Callbacks *callbacks) {
 
   std::string role_arn = default_role_arn_;
+ 
   // If role_arn_arg is present, use that, otherwise use env
   if (role_arn_arg.has_value()) {
     role_arn = role_arn_arg.value();
   }
 
+  std::string role_arn_lookup = role_arn;
+  if (disable_role_chaining) {
+    // if disable_role_chaining is set on an upstream we need a distinction 
+    // so that we can serve both chained and non-chained as needed
+    role_arn_lookup = "no-chain" + role_arn;
```

Comment:
Super nit: Should we put a separator here?

Reply:
how do you feel about "no-chain-"